4_predict/src/select_features.R

```diff
@@ -0,0 +1,273 @@
+drop_high_corr_attrs <- function(features, drop_columns = NULL, drop_exact = NULL,
+                                 dynamic_contains,
+                                 threshold_corr, first_drop_var_prefix = NULL,
+                                 cor_method){
+  #' 
+  #' @description Function to remove attributes that are undesired or are 
+  #' highly correlated with other attributes. 
+  #'
+  #' @param features table of features (columns) for each prediction point (rows).
+  #' @param drop_columns character vector of column names to remove from features. 
+  #' This is passed to dplyr::contains()
+  #' @param drop_exact character vector of exact column names to drop.
+  #' This is passed to dplyr::select()
+  #' @param dynamic_contains character vector of column name formats that indicate
+  #' an attribute is dynamic. This is passed to dplyr::contains()
+  #' @param threshold_corr correlation threshold used to drop attributes.
+  #' Absolute value of correlation is used.
+  #' @param first_drop_var_prefix if not NULL, an initial screening will be made
+  #' and highly correlated features with the specified prefix will be dropped. 
+  #' Example prefix for NHD features: "CAT", "ACC", or "TOT"
+  #' @param cor_method correlation method for cor function.
+  #' 
+  #' @return character vector of features that are retained
+  
+  if (!is.null(drop_columns)){
+    features <- select(features, !contains(drop_columns))
+  }
+  if (!is.null(drop_exact)){
+    features <- select(features, !all_of(drop_exact))
+  }
+  
+  #Drop rows with NAs
+  features <- drop_na(features)
```

Comment:
> Does this drop the entire row if a feature is NA?

Yes, it drops the whole row. In this case, that amounts to dropping data before 1985 water year. 
```
max(features[which(is.na(features), arr.ind = T)[,1], c(1,unique(which(is.na(features), arr.ind = T)[,2]))]$Date)
[1] "1984-09-29"
```
I kept it general in this function in case new attributes are added that have NAs that are not handled before this function.

> What if you end up not using the feature that yields the NA value?

For this function, the correlation is only checked for rows with all feature data available. For modeling, rows with NAs are still available to use if the features are not selected for use. This function returns the column names of features to use.